superset/utils/decorators.py

```diff
@@ -111,7 +112,9 @@ def wrapped(*args: Any, **kwargs: Any) -> Any:
     return decorate
 
 
-def on_security_exception(self: Any, ex: Exception) -> Response:
+def on_security_exception(self: Any, ex: Exception) -> Union[Response, werkz_Response]:
+    if not g.user or not utils.get_user_id():
+        return redirect(appbuilder.get_url_for_login)
```

Comment:
Should probably just return the following as the caller could be expecting json response and that redirect will end up returning html. Not sure if making this change will actually still fix the issue, but this probably isn't the place to fix that specific issue as it's a general decorator used by lots of routes. 
```suggestion
        return self.response(401, **{"message": utils.error_msg_from_exception(ex)})
```

Reply:
Actually, you might be right here, looks like this is only ever used in one place

Reply:
hmm, looks like the only place `check_dashboard_access` is actually used, this default arg is overridden with a lambda that pretty much does the exact same thing 🤔 

https://github.com/apache/superset/blob/ed11123709e51f7a8616f089a3d27b3e3159f843/superset/views/core.py#L1822

Reply:
Not sure how but when I tested it is calling on_security_exception, the default callable in case of DashboardAccessDeniedError occurred.  Do you think instead of making changes here  in on_security_exception we should look into why lambda is not working? 🤔